include/sys/__assert.h

```diff
@@ -20,12 +20,38 @@
 #define __ASSERT_ON 0
 #endif
 
+#if defined(CONFIG_ASSERT_VERBOSE)
+#define __ASSERT_PRINT(fmt, ...) printk(fmt, ##__VA_ARGS__)
+#else /* CONFIG_ASSERT_VERBOSE */
+#define __ASSERT_PRINT(fmt, ...)
+#endif /* CONFIG_ASSERT_VERBOSE */
+
 #ifdef CONFIG_ASSERT_NO_FILE_INFO
 #define __ASSERT_FILE_INFO ""
+#define __ASSERT_LINE_INFO 0
```

Comment:
What would be the benefit of removing the line number from the message and replacing it with 0? (Are you hoping the printk string will be optimized into a constant string with less chars?)

Reply:
I wanted to not have all the numbers in of the asserts in the final image. But it seems that this doesn't make any difference as now there are just a whole lot of zeroes instead. I would have to do more to optimize it away it seems. Which would probably include removing the arguments from the assert_post_action function.

Reply:
Completely removing file and line gives me around 3656 bytes saved for the shell. It's roughly the same for the peripheral sample. Around 3k.